v21.1/alembic.md

```diff
@@ -0,0 +1,569 @@
+---
+title: Migrate CockroachDB Schemas with Alembic
+summary: Learn how to use Alembic with a CockroachDB cluster.
+toc: true
+---
+
+This page walks you through a series of simple database schema changes using the [Alembic](https://alembic.sqlalchemy.org/en/latest/) schema migration module with a simple Python application built on SQLAlchemy and CockroachDB.
+
+For a detailed tutorial about using Alembic, see the [the Alembic documentation site](https://alembic.sqlalchemy.org/en/latest/tutorial.html).
+
+For information about specific migration tasks, see Alembic's [Cookbook](https://alembic.sqlalchemy.org/en/latest/cookbook.html).
+
+## Before you begin
+
+Before you begin the tutorial, [install CockroachDB](install-cockroachdb.html).
+
+## Step 1. Start a cluster and create a database
+
+1. Start a [demo cluster](cockroach-demo.html):
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ cockroach demo --no-example-database
+    ~~~
+
+    This command creates a virtual cluster and opens a SQL shell to that cluster.
+
+    {{site.data.alerts.callout_info}}
+    Leave this terminal window open for the duration of the tutorial. Closing the window will destroy the cluster and erase all data in it.
+    {{site.data.alerts.end}}
+
+1. Create the `bank` database:
+
+    {% include copy-clipboard.html %}
+    ~~~ sql
+    > CREATE DATABASE bank;
+    ~~~
+
+## Step 2. Get the application code
+
+1. Open a new terminal, and clone the [`simple-crud-python-sqlalchemy`](https://github.com/cockroachlabs/simple-crud-python-sqlalchemy) GitHub repository:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ git clone git@github.com:cockroachlabs/simple-crud-python-sqlalchemy.git
+    ~~~
+
+## Step 3. Install and initialize Alembic
+
+1. Navigate to the `simple-crud-python-sqlalchemy` project directory, and run the following commands to create and start a virtual environment:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ python3 -m venv env
+    ~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ source env/bin/activate
+    ~~~
+
+1. Install the `alembic`, [`sqlalchemy-cockroachdb`](https://github.com/cockroachdb/sqlalchemy-cockroachdb), and [`psycopg2`](https://github.com/psycopg/psycopg2/) modules to the virtual environment:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ pip install sqlalchemy-cockroachdb psycopg2 alembic
```

Comment:
Relatively new users may stumble on `pip install psycopg2` failing because their machine does not have the prerequisites to build psycopg2 from source (e.g., `sudo apt install python3-dev libpq-dev` on Ubuntu). `pip install psycopg2-binary` might be a safer choice, especially for a tutorial.

Reply:
Good point! Updated.